address.go

```diff
@@ -375,3 +483,82 @@ func (a *AddressPubKey) AddressPubKeyHash() *AddressPubKeyHash {
 func (a *AddressPubKey) PubKey() *btcec.PublicKey {
 	return a.pubKey
 }
+
+// AddressSegwit is an Address for a pay-to-witness-pubkey-hash (P2WPKH) or
```

Comment:
I originally did this, but realized that these two would restrict us to version `0` address types (I might be wrong here, so correct me if I am). In case we encounter a non-version 0 address, we _could_ fail parsing it, but that would make the implementation non-forwards compatible. We could also create a third type to hold non-zero types, but maybe that is unnecessary complex?